modules/local/failurecheck/main.nf

```diff
@@ -0,0 +1,56 @@
+process FAILURE_CHECK {
+    tag "failure_check"
+    label 'process_single'
+
+    input:
+    val fastp_fail
+    val kraken_fail
+    val bracken_fail
+
+    output:
+    path("failures_report.csv"), emit: failures_report
+
+    exec:
+    def processedIDs = [:]
```

Comment:
Given the way you are using `processedIDs` (to keep track of IDs), you can declare this as a [Set](https://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Set.html) data structure instead of a Map (https://stackoverflow.com/a/27848434). That is:

```groovy
def processedIDs = [] as Set
```

You can then use it like:

```groovy
// Add id to set
processedIDs.add(id)
```

And 

```groovy
// If id not in set
if (!id in processedIDs) {
}
```

Reply:
Love this! Thanks for the suggestion: [f4f0b79](https://github.com/phac-nml/speciesabundance/pull/20/commits/f4f0b79a0e15a0acb74af1c601469593c31788f2)